simd/2021/SIMD-05-14.md

```diff
@@ -0,0 +1,166 @@
+
+PP: worked with open power a bit, vector of conditions are hard to manage
+
+JL: not based on open power SIMD support, independent thing. It is based on scalar instruction, prefix, runs in a loop
+
+PP: cool. What kind of performance do you see / expect to see from this
+
+JL: for initial implementation, we set it up with 128 bit wide SIMD execution units, we’re using an older process to keep costs down, we’ll probably end up limiting to 800 mhz. 50 gflops. 4-core processor.
+
+PP: how will this compare to something else, say when you take ARM or Atom.
+
+JL: we have been comparing to Raspberry Pi GPU. approximately equivalent
+
+FL: q about vector length, what is the goal of the vector length register, is it for loop prologues, or just to set up preferred length of computation
+
+JL: vector length register tells the processor that you don’t have to execute all 64 lanes, just 5 or whatever you set it to. You won’t have to do 16 clock cycles when you only have to do 2, if the processor can only do 4 lanes at a time
+
+FL: what happens when you reduce VL to data that is out of bounds in registers
+
+JL: we have instruction for setting VL, kind of like simpleV one, it is just an immediate to instruction, compiler will set it based on number of registers allocated for the instruction
+
+FL: not clear
+
+JL: instruction has an immediate that says maximum length allocated, another arg that is an int arg, will use the smaller one as the vector length, the programmer can have a loop that goes back to do the next X elements. Language level help for loop strip mining. Better than traditional SIMD instructions, we don’t have huge gigantic complicated code for cleaning up the extra ends.
+
+FL: assuming we want to do a sum of an array, what you would do with predicate and mask is, each iteration, the full length, last iteration you mask the load of the last elements, with set length like in RISC-V V, last element of accumulator mandated to stay, thus when you reduce the elements of the register, you reduce all the elements, and not only the remaining ones. Is this the same in SimpleV?
+
+JL: yes, if you have VL set to less than max length, it won’t modified any of the element past the vector length. Perhaps unlike RISC-V V, not sure if they require elements to be unmodified. If you run the configure vector instruction, it throws away all the stuff.
+
+FL: RISC-V elements after vec length mandated to remain the same value
+
+JL: good, then it is the same
+
+JW: that is configurable, that is a mode to choose between undisturbed and don’t-care
+
+PP: when you came to say that you are working on open source SOC, it was unclear what it means, looking forward to see progress.
+
+JL (chat): iirc here's the SimpleV link https://libre-soc.org/openpower/sv/overview/
+
+### Masks (Florian Lemaitre)
+
+[Slides](https://nuage.lip6.fr/s/oFfcdHeTm7CPcms)
```

Comment:
I suggest to upload the slides in github. This link is bound to my lab account, and my contract is close to an end, so I cannot guarantee it will remain for a long time.

Reply:
Done, thanks!